reactioncommerce/reaction
# feat: add updateProduct and updateProductVariant mutations for 3.0

This PR takes the GQL mutation changes from https://github.com/reactioncommerce/reaction/pull/5686/files, and PR's them into `release-3.0.0`. This PR does not include any of the client-side/meteor changes, as those would all be made into the `reaction-admin` project in `3.0.0`.

From #5686:

Resolves parts of #5679   
Impact: **minor**  
Type: **feature|refactor**

## Issue
As part of our effort to demeteorize the app, we need to convert remaining meteor methods to GQL mutations/queries.

## Solution

This PR adds the following mutations: `updateProduct` and `updateProductVariant`

Instead of creating a second mutation to clone the `products/toggleVisibility` meteor method, I've just confirmed that `toggleVisibility` works in this new mutation.

## Breaking changes
Any custom code using any of the above meteor methods will need to be refactored to use the new GQL methods.

## Testing
1. Start the app.
1. See that you can perform the following actions:
- Update fields on a product
- Update fields on a product variant

```graphql
mutation updateProductMutation {
	updateProduct(input: {
    productId: "PRODUCT_ID"
    shopId: "SHOP_ID"
    product: {
      title: "My other title product"
			isVisible: false
    }
  }) {
    product {
      _id
      title
      isVisible
    }
  }
}

mutation updateProductVariantMutation {
	updateProductVariant(input: {
    variantId: "VARIANT_ID"
    shopId: "SHOP_ID"
    variant: {
      title: "My other title"
    	attributeLabel: "Something"
			isVisible: false
    }
  }) {
    variant {
      _id
      title
      attributeLabel
      isVisible
    }
  }
}

```
